realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -319,14 +324,16 @@ public void clear() {
      */
     @Override
     public E remove(int location) {
+        E removedItem;
         if (managedMode) {
             checkValidView();
-            E removedItem = get(location);
+            removedItem = get(location);
             view.remove(location);
-            return removedItem;
         } else {
-            return nonManagedList.remove(location);
+            removedItem = nonManagedList.remove(location);
         }
+        modCount++;
```

Comment:
Doesn't seem to be big issue, but `ArrayList.remove()` will do `modCount++`. Same with others.